OfficeAddin/05 Using modern JavaScript/Demos/01 Office Add-in using React/src/utils/ExcelTableUtil.tsx

```diff
@@ -2,167 +2,167 @@
+      return new Promise(async (resolve, reject) => {
+        this.ensureTable(true).then(
+          async (tableRef: Excel.Table) => {
             await Excel.run(async context => {
-              // Get column range by column name
               const sheet = context.workbook.worksheets.getActiveWorksheet();
+              // Add the new row
               tableRef = sheet.tables.getItem(this.tableName);
-              var colRange = tableRef.columns.getItem(column).getDataBodyRange().load("values");
-                        
-              // Sync to populate proxy objects with data from Excel
-              return context.sync().then(async () => {
-                let data: string[] = [];
-                for (let i = 0; i < colRange.values.length; i++) {
-                  data.push(colRange.values[i].toString());
-                }
-                resolve(data);
+              tableRef.rows.add(null, [data]);
+              // Autofit columns and rows if your Office version supports the API.
+              if (Office.context.requirements.isSetSupported('ExcelApi', 1.2)) {
+                sheet.getUsedRange().format.autofitColumns();
+                sheet.getUsedRange().format.autofitRows();
+              }
+              sheet.activate();
+              return context.sync().then(() => {
+                resolve();
               });
             }).catch(err => {
               reject(err);
             });
+          },
+          err => {
+            reject(err);
           }
-        },
-        err => {
-          reject(err);
-        }
-      );
-    });
-  }
+        );
+      });
+    }
   
-  // Deletes a column based by row index
-  deleteRow = async (index: number) => {
-    return new Promise(async (resolve, reject) => {
-      this.ensureTable(true).then(
-        async (tableRef: Excel.Table) => {
-          await Excel.run(async context => {
-            const range = tableRef.rows.getItemAt(index).getRange();
-            range.delete(Excel.DeleteShiftDirection.up);
-            return context.sync().then(async () => {
-              resolve();
+    // Gets data for a specific named column
+    getColumnData = async (column: string) => {
+      return new Promise(async (resolve, reject) => {
+        this.ensureTable(false).then(
+          async (tableRef: Excel.Table) => {
+            if (tableRef == null) {
+              resolve([]);
+            } else {
+              await Excel.run(async context => {
+                // Get column range by column name
+                const sheet = context.workbook.worksheets.getActiveWorksheet();
+                tableRef = sheet.tables.getItem(this.tableName);
+                var colRange = tableRef.columns.getItem(column).getDataBodyRange().load("values");
+                          
+                // Sync to populate proxy objects with data from Excel
+                return context.sync().then(async () => {
+                  let data: string[] = [];
```

Comment:
This added string[] type is not in the lab code that you copy in for the Angular exercise. Although it seems to run fine. Just pointing out a difference.

Reply:
Updating to match